src/lib.rs

```diff
@@ -60,12 +54,22 @@ impl Firebase {
     ///   will be returned.
     /// - If a url cannot be parsed into a valid url then a ```Err(ParseError::Parser(url::ParseError)```
     ///   will be returned.
-    pub fn from_url(url: &Url) -> Result<Self, ParseError> {
-        let url = url.clone();
+    pub fn from_url(mut url: Url) -> Result<Self, ParseError> {
         try!( unwrap_path(&url) );
 
+        if url.scheme() != "https" {
+            return Err(ParseError::UrlIsNotHTTPS);
+        }
+
+        {
+            // Append initial .json
+            let mut last = url.path_segments().expect("last segment").last().unwrap().to_string();
+            let mut path = url.path_segments_mut().expect("path segments");
```

Comment:
This is meant to be the Firebase root, what happens if I type in `myapp.firebase.com` ? is `https://myapp.firebase.com/.json` a valid firebase url?